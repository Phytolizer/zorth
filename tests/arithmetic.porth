include "std.porth"

// add
34 35 + print

"------------------------------\n" write

// subtract
500 80 - print

"------------------------------\n" write

// multiply
42 10 * print

"------------------------------\n" write

// divide
69000 1000 / print

"------------------------------\n" write

// modulo
5 100 mod print
104 100 mod print
