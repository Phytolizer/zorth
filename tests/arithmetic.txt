69
------------------------------
420
------------------------------
5
4
------------------------------
420
