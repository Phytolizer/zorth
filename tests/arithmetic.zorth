// add
34 35 + print

"------------------------------\n" 1 1 syscall3 drop

// subtract
500 80 - print

"------------------------------\n" 1 1 syscall3 drop

// modulo
5 100 mod print
104 100 mod print

"------------------------------\n" 1 1 syscall3 drop

// multiply
42 10 * print
