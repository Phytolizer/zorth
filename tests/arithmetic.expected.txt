69
------------------------------
420
------------------------------
420
------------------------------
5
4
